pkg/api/apis/operators/v1alpha1/catalogsource_types.go

```diff
@@ -18,16 +18,43 @@ const (
 type SourceType string
 
 const (
-	SourceTypeInternal  SourceType = "internal"
+	// SourceTypeInternal (deprecated) specifies a CatalogSource of type SourceTypeConfigmap
+	SourceTypeInternal SourceType = "internal"
+
+	// SourceTypeConfigmap specifies a CatalogSource that generates a configmap-server registry
 	SourceTypeConfigmap SourceType = "configmap"
-	SourceTypeGrpc      SourceType = "grpc"
+
+	// SourceTypeGrpc specifies a CatalogSource that can use an operator registry image to generate a
+	// registry-server or connect to a pre-existing registry at an address.
+	SourceTypeGrpc SourceType = "grpc"
 )
 
 type CatalogSourceSpec struct {
+	// SourceType is the type of source
 	SourceType SourceType `json:"sourceType"`
-	ConfigMap  string     `json:"configMap,omitempty"`
-	Image      string     `json:"image,omitempty"`
-	Secrets    []string   `json:"secrets,omitempty"`
+
+	// ConfigMap is the name of the ConfigMap to be used to back a configmap-server registry.
+	// Only used when SourceType = SourceTypeConfigmap or SourceTypeInternal.
+	// +Optional
+	ConfigMap string `json:"configMap,omitempty"`
+
+	// Address an address that OLM can use to connect to a pre-existing registry.
```

Comment:
nit: "Address is a host that OLM...". Since one e2e failed, maybe it's worth fixing.